mlflow/openai/__init__.py

```diff
@@ -589,6 +594,20 @@ def log_model(
 
 
 def _load_model(path):
+    model_file_path = os.path.dirname(path)
+    if os.path.exists(model_file_path):
+        mlflow_model = Model.load(model_file_path)
+        if (
+            mlflow_model.model_id
+            # only set the active model if the model is not set by the user
+            and not _is_active_model_id_set_by_user()
+            and get_active_model_id() != mlflow_model.model_id
```

Comment:
```suggestion
            and m := get_active_model_context()
            and m.set_by_user
            and m.model_id == mlflow_model.model_id
```

can we fetch the active model context and compare its attributes?

Reply:
> get_active_model_id() also reads from env variable, we should respect that value as well

Can we include this logic in `ActiveModelContext`?

Reply:
Ah yea let me update ActiveModelContext class